docs/samples/Microsoft.ML.Samples/Dynamic/Transforms/TimeSeries/DetectChangePointBySsaStream.cs

```diff
@@ -0,0 +1,138 @@
+                new TimeSeriesData(3),
+                new TimeSeriesData(4),
+
+                new TimeSeriesData(0),
+                new TimeSeriesData(1),
+                new TimeSeriesData(2),
+                new TimeSeriesData(3),
+                new TimeSeriesData(4),
+
+                new TimeSeriesData(0),
+                new TimeSeriesData(1),
+                new TimeSeriesData(2),
+                new TimeSeriesData(3),
+                new TimeSeriesData(4),
+            };
+
+            // Convert data to IDataView.
+            var dataView = ml.Data.LoadFromEnumerable(data);
+
+            // Setup SsaChangePointDetector arguments
+            var inputColumnName = nameof(TimeSeriesData.Value);
+            var outputColumnName = nameof(ChangePointPrediction.Prediction);
+
+            // Train the change point detector.
+            ITransformer model = ml.Transforms.DetectChangePointBySsa(outputColumnName, inputColumnName, 95, 8, TrainingSize, SeasonalitySize + 1).Fit(dataView);
+
+            // Create a prediction engine from the model for feeding new data.
+            var engine = model.CreateTimeSeriesPredictionFunction<TimeSeriesData, ChangePointPrediction>(ml);
+
+            // Start streaming new data points with no change point to the prediction engine.
+            Console.WriteLine($"Output from ChangePoint predictions on new data:");
+            Console.WriteLine("Data\tAlert\tScore\tP-Value\tMartingale value");
+            
+            // Output from ChangePoint predictions on new data:
+            // Data    Alert   Score   P-Value Martingale value
+
+            for (int i = 0; i < 5; i++)
+                PrintPrediction(i, engine.Predict(new TimeSeriesData(i)));
+
+            // 0       0      -1.01    0.50    0.00
+            // 1       0      -0.24    0.22    0.00
+            // 2       0      -0.31    0.30    0.00
+            // 3       0       0.44    0.01    0.00
+            // 4       0       2.16    0.00    0.24
+
+            // Now stream data points that reflect a change in trend.
+            for (int i = 0; i < 5; i++)
+            {
+                int value = (i + 1) * 100;
+                PrintPrediction(value, engine.Predict(new TimeSeriesData(value)));
+            }
+            // 100     0      86.23    0.00    2076098.24
+            // 200     0     171.38    0.00    809668524.21
+            // 300     1     256.83    0.01    22130423541.93    <-- alert is on, note that delay is expected
+            // 400     0     326.55    0.04    241162710263.29
+            // 500     0     364.82    0.08    597660527041.45   <-- saved to disk
+
+            // Now we demonstrate saving and loading the model.
+
+            // Save the model that exists within the prediction engine.
+            // The engine has been updating this model with every new data point.
+            var modelPath = "model.zip";
+            using (var stream = File.Create(modelPath))
```

Comment:
Should we use a different kind of Stream than a file? That way this sample is different than the other one that also writes to a file.

Reply:
Good idea!